src/DeviceDataStreamValues.tsx

```diff
@@ -0,0 +1,295 @@
+        <tbody>
+          {data.map((tree) => (
+            <tr key={tree.timestamp}>
+              {objectProperties.map((property) => (
+                <td key={property}>{formatAstarteData(tree.value[property])}</td>
+              ))}
+              <td>{new Date(tree.timestamp).toLocaleString()}</td>
+            </tr>
+          ))}
+        </tbody>
+      </Table>
+    </>
+  );
+};
+
+interface InterfaceDataProps {
+  interfaceData: AstarteDatastreamIndividualData[] | AstarteDatastreamObjectData[];
+  aggregation: AstarteInterface['aggregation'];
+}
+
+const InterfaceData = ({ interfaceData, aggregation }: InterfaceDataProps): React.ReactElement => {
+  if (interfaceData.length === 0) {
+    return <p>No data in the selected timeframe.</p>;
+  } else if (aggregation === 'individual') {
+    return <IndividualDatastreamTable data={interfaceData as AstarteDatastreamIndividualData[]} />;
+  }
+  return <ObjectDatastreamTable data={interfaceData as AstarteDatastreamObjectData[]} />;
+};
+
+export default (): React.ReactElement => {
+  const { interfaceName = '', deviceId = '', interfaceMajor } = useParams();
+  const astarte = useAstarte();
+  const [requestingData, setRequestingData] = useState(false);
+  const [fetchedData, setFetchedData] = useState<AstarteInterfaceValues | null>(null);
+  const [formAlerts, formAlertsController] = useAlerts();
+  const [interfaceData, setInterfaceData] = useState<AstarteInterface | null>(null);
+  const [interfacePaths, setInterfacePaths] = useState<string[]>([]);
+  const [selectedPath, setSelectedPath] = useState<string>('');
+
+  const fetchData = async (path: string, since?: string, to?: string) => {
+    setRequestingData(true);
+    setFetchedData(null);
+    astarte.client
+      .getDeviceData({
+        deviceId,
+        interfaceName,
+        path,
+        since,
+        to,
+      })
+      .then((data) => {
+        setSelectedPath(path);
+        setFetchedData(data);
+      })
+      .catch((err) => {
+        formAlertsController.showError(
+          `Could not fetch data to interface: ${err.response.data.errors.detail}`,
+        );
+        setFetchedData([]);
+      })
+      .finally(() => {
+        setRequestingData(false);
+      });
+  };
+
+  const transformedData =
+    interfaceData && fetchedData
+      ? interfaceData.aggregation === 'individual'
+        ? transformIndividualDatastreamInterfaceValues(selectedPath, fetchedData)
+        : transformAggregatedDatastreamInterfaceValues(selectedPath, fetchedData)
+      : [];
+
+  useEffect(() => {
+    if (interfaceMajor) {
+      const major = parseInt(interfaceMajor, 10);
+      astarte.client
+        .getInterface({ interfaceName, interfaceMajor: major })
+        .then((interfaceData) => {
+          setInterfaceData(interfaceData);
+          const session = loadSession();
+          if (session && session.paths) {
+            setInterfacePaths(session.paths);
+          }
+        });
+    }
+  }, []);
```

Comment:
`loadSession` is not a clever idea: if the user wants to look at the data of different devices using multiple browser tabs, the same React state `interfacePaths` gets randomly overwritten with a list of paths that are correct for one device, but not for the other devices.

Since you're using the result of  `dataTreeNode.getLeaves().map((data) => data.endpoint)` to retrieve the list of interface path in a simplified way, I suggest you used that method directly here in this component, without passing through the Astarte Manager and `saveInterfacePathsInSession` / `loadSession`.
Something like this should be enough:

```ts
const deviceDataFetcher = useFetch(() =>
  astarte.client.getDeviceDataTree({
    deviceId,
    interfaceName,
  }),
);

const interfacePaths = useMemo(() => {
  if (deviceDataFetcher.value == null) {
    return [];
  }
  return deviceDataFetcher.value.toLinearizedData().map((data) => data.endpoint);
}, [deviceDataFetcher.value]);
```